pkg/controller/account/ec2.go

```diff
@@ -142,25 +143,39 @@ func CreateEC2Instance(reqLogger logr.Logger, client awsclient.Client, ami strin
 		}
 		// Specify the details of the instance that you want to create.
 		runResult, runErr := client.RunInstances(&ec2.RunInstancesInput{
-			// An Amazon Linux AMI ID for t2.micro instances in the us-west-2 region
 			ImageId:      aws.String(ami),
 			InstanceType: aws.String(awsInstanceType),
 			MinCount:     aws.Int64(1),
 			MaxCount:     aws.Int64(1),
 		})
-		if runErr == nil {
-			instanceID = *runResult.Instances[0].InstanceId
-			break
-		}
 
-	}
+		// Assign instance id
+		instanceID = *runResult.Instances[0].InstanceId
+
+		// Return on unexpected errors:
+		if runErr != nil {
```

Comment:
I think somehow this error should be tracked and returned after the loop or we should have a counter that says the number of errors outside of the loop and if it is greater then 1 return the error.

Reply:
We already log all errors that are not expected:
```
reqLogger.Error(runErr,
  fmt.Sprintf(`Failed while trying to create EC2 instance, 
  AWS Error Code: %s, 
  AWS Error Message: %s`,
  aerr.Code(),
  aerr.Message()))
```